Makefile

```diff
@@ -31,40 +32,59 @@ all: docker
 gobuild:
 	./scripts/build false
 
+# create output directories
+.out-stamp:
+	mkdir -p ./out/test-artifacts ./out/cni-plugins
+	touch .out-stamp
+
+out: .out-stamp
+
 # Basic go build
 static:
 	./scripts/build
 
+BUILDER_IMAGE="amazon/amazon-ecs-agent-build:make"
+.builder-image-stamp: scripts/dockerfiles/Dockerfile.build
+	@docker build -f scripts/dockerfiles/Dockerfile.build -t $(BUILDER_IMAGE) .
+	touch .builder-image-stamp
+
+builder-image: .builder-image-stamp
```

Comment:
Seems like `builder-image` is a nop, why is it needed?

Reply:
Its not a `NOP`. It builds a dockerfile thats shared between multiple targets.

Reply:
Just kidding -- this is a nop. Its just a more meaningful name that other targets depend on.